state-sync/state-sync-v2/state-sync-driver/src/storage_synchronizer.rs

```diff
@@ -97,15 +98,34 @@ pub struct StorageSynchronizer {
     // The number of storage data chunks pending execute/apply, or commit
     pending_data_chunks: Arc<AtomicU64>,
 
+    // An optional runtime on which to spawn the storage synchronizer threads
+    runtime: Option<Handle>,
+
     // The channel through which to notify the state snapshot receiver of new data chunks
     state_snapshot_notifier: Option<mpsc::Sender<StorageDataChunk>>,
 
     // The writer to storage (required for account state syncing)
     storage: Arc<dyn DbWriter>,
 }
 
-impl StorageSynchronizer {
-    pub fn new<ChunkExecutor: ChunkExecutorTrait + 'static>(
+// TODO(joshlind): the compiler isn't happy with automatically deriving this...
```

Comment:
lol.. any idea why?

Reply:
Yeah, it's because deriving Clone only really works for simple cases. There's a bunch of cases where it doesn't, e.g.,: https://github.com/rust-lang/rust/issues/64417 and https://github.com/rust-lang/rust/issues/26925. In our case, we have to manually define it.